### PR TITLE
[Enhance] Support reading class_weight from file in loss functions to help MMDet3D

### PR DESCRIPTION
**Pull request by Wuziyi616**
Hi, I am Ziyi, a developer from MMDet3D. Currently we are implementing 3D segmentation models, which require using loss functions from MMSeg (e.g. CrossEntropy). It's very common to use class_weight in 3D Seg, and in some scenarios we need to load class_weight from files. So I modify the code and support this feature. Hope you can approve it. Thanks.

### CODE REVIEW

File: mmseg/models/losses/cross_entropy_loss.py [hunk 1]
```diff
@@ -172,6 +172,28 @@ def __init__(self,
...
+        import mmcv
+
+        if isinstance(class_weight, str):
+            # take it as a file path
+            if class_weight.endswith('.npy'):
+                class_weight = np.load(class_weight)
+            else:
+                file_format = class_weight.split('.')[-1]
+                assert file_format in ['json', 'yaml', 'pkl'], \
+                    f'unsupported class_weight file type {file_format}'
```
Comment (xvjiarui) [inline mmseg/models/losses/cross_entropy_loss.py:192]: This part is already handled by MMCV [here](https://github.com/open-mmlab/mmcv/blob/master/mmcv/fileio/io.py#L37)

File: mmseg/models/losses/cross_entropy_loss.py [hunk 2]
```diff
@@ -172,6 +172,28 @@ def __init__(self,
         else:
             self.cls_criterion = cross_entropy
 
+    def _get_class_weight(self, class_weight):
```
Comment (xvjiarui) [inline mmseg/models/losses/cross_entropy_loss.py:175]: Could we make this an utility function that could be shared by all losses?

File: mmseg/models/losses/cross_entropy_loss.py (hunk 1 above)
Comment (Wuziyi616) [reply mmseg/models/losses/cross_entropy_loss.py]: Great to hear.

File: mmseg/models/losses/cross_entropy_loss.py (hunk 2 above)
Comment (Wuziyi616) [reply mmseg/models/losses/cross_entropy_loss.py]: Yes, good idea!

**Comment by Wuziyi616** [comment]
This PR is ready for review @xvjiarui. Thanks!

**Review by xvjiarui — APPROVED** [review]
